internal/app/starter/master_linux.go

```diff
@@ -11,129 +11,131 @@ import (
 	"net"
 	"os"
 	"os/signal"
-	"runtime"
 	"syscall"
-	"time"
 
 	"github.com/sylabs/singularity/internal/pkg/runtime/engines/engine"
 	"github.com/sylabs/singularity/internal/pkg/sylog"
 	"github.com/sylabs/singularity/internal/pkg/util/mainthread"
 )
 
-// Master initializes a runtime engine and runs it
-func Master(rpcSocket, masterSocket int, isInstance bool, containerPid int, e *engine.Engine) {
-	var fatal error
-	var status syscall.WaitStatus
-	fatalChan := make(chan error, 1)
-
-	// we could receive signal from child with CreateContainer call so we
-	// set the signal handler earlier to queue signals until MonitorContainer
-	// is called to handle them
-	signals := make(chan os.Signal, 1)
-	signal.Notify(signals)
+func createContainer(rpcSocket int, containerPid int, e *engine.Engine, fatalChan chan error) {
+	comm := os.NewFile(uintptr(rpcSocket), "rpc-socket")
+	if comm == nil {
+		fatalChan <- fmt.Errorf("bad RPC socket file descriptor")
+		return
+	}
+	rpcConn, err := net.FileConn(comm)
+	comm.Close()
+	if err != nil {
+		fatalChan <- fmt.Errorf("failed to copy unix socket descriptor: %s", err)
+		return
+	}
 
-	ppid := os.Getppid()
+	err = e.CreateContainer(containerPid, rpcConn)
```

Comment:
Why there is no `runtime.LockOSThread` any more? 
In comments above I see you had some discussion with @cclerget in slack, was this discussed as well?

Reply:
@sashayakovtseva This is incorrect, I did not discuss this point. The initial comment was that a reviewer (I do not remember who and it does not matter) did not like the naming I selected. @cclerget made a pass over the code to cleanup/ease the implementation of unit tests.
But are you asking here? The addition of comments?